magentoerpconnect/sale.py

```diff
@@ -681,13 +681,20 @@ def _clean_magento_items(self, resource):
         child_items = {}  # key is the parent item id
         top_items = []
 
+        prefix = self.env['ir.config_parameter'].get_param('outlet_product_prefix')
         is_pre_order = resource.get('extension_attributes', {}).get(
             'is_pre_order')
         # Group the childs with their parent
         for item in resource['items']:
             # Pre-order customizations
             if item.get('sku') and item['sku'].startswith('PRE-ORDER'):
                 item['sku'] = 'PRE-ORDER'
+            # Outlet products mapping
+            elif item.get('sku') and item.get('sku').startswith(prefix+'-'):
+                sufix = item.get('sku').split('-')[-1]
+                item['outlet_route_hint'] = sufix
+                item['sku'] = item.get('sku').replace(
+                    prefix+'-','').replace('-'+sufix,'')
```

Comment:
You might replace too much, for instance if the suffix = '30' and the product = 'VM01-301'. Try `[len(prefix) + 1:-(len(suffix) + 1)]` instead.

Reply:
that is a nice solution